bpfman/src/multiprog/xdp.rs

```diff
@@ -285,59 +310,83 @@ impl XdpDispatcher {
         Ok(())
     }
 
-    fn save(&self) -> Result<(), BpfmanError> {
-        debug!(
-            "XdpDispatcher::save() for if_index {}, revision {}",
-            self.if_index, self.revision
-        );
-        let path = format!("{RTDIR_XDP_DISPATCHER}/{}_{}", self.if_index, self.revision);
-        serde_json::to_writer(&fs::File::create(path).unwrap(), &self)
-            .map_err(|e| BpfmanError::Error(format!("can't save state: {e}")))?;
-        Ok(())
-    }
-
-    pub fn load(if_index: u32, revision: u32) -> Result<Self, anyhow::Error> {
-        debug!("XdpDispatcher::load() for if_index {if_index}, revision {revision}");
-        let path = format!("{RTDIR_XDP_DISPATCHER}/{if_index}_{revision}");
-        let file = fs::File::open(path)?;
-        let reader = BufReader::new(file);
-        let prog = serde_json::from_reader(reader)?;
-        // TODO: We should check the bpffs paths here to for pinned links etc...
-        Ok(prog)
-    }
-
     pub(crate) fn delete(&self, full: bool) -> Result<(), BpfmanError> {
+        let if_index = self.get_ifindex()?;
+        let revision = self.get_revision()?;
         debug!(
             "XdpDispatcher::delete() for if_index {}, revision {}",
-            self.if_index, self.revision
-        );
-        let path = format!("{RTDIR_XDP_DISPATCHER}/{}_{}", self.if_index, self.revision);
-        fs::remove_file(path)
-            .map_err(|e| BpfmanError::Error(format!("unable to cleanup state: {e}")))?;
-
-        let path = format!(
-            "{RTDIR_FS_XDP}/dispatcher_{}_{}",
-            self.if_index, self.revision
+            if_index, revision
         );
+        ROOT_DB.drop_tree(self.db_tree.name()).map_err(|e| {
+            BpfmanError::DatabaseError(
+                format!(
+                    "unable to drop xdp dispatcher tree {:?}",
+                    self.db_tree.name()
+                ),
+                e.to_string(),
+            )
+        })?;
+
+        let path = format!("{RTDIR_FS_XDP}/dispatcher_{}_{}", if_index, revision);
         fs::remove_dir_all(path)
             .map_err(|e| BpfmanError::Error(format!("unable to cleanup state: {e}")))?;
         if full {
-            let path_link = format!("{RTDIR_FS_XDP}/dispatcher_{}_link", self.if_index);
+            let path_link = format!("{RTDIR_FS_XDP}/dispatcher_{}_link", if_index);
             fs::remove_file(path_link)
                 .map_err(|e| BpfmanError::Error(format!("unable to cleanup state: {e}")))?;
         }
         Ok(())
     }
 
-    pub(crate) fn if_name(&self) -> String {
-        self.if_name.clone()
+    pub(crate) fn set_revision(&mut self, revision: u32) -> Result<(), BpfmanError> {
+        sled_insert(&self.db_tree, "revision", &revision.to_ne_bytes())
+    }
+
+    pub(crate) fn get_revision(&self) -> Result<u32, BpfmanError> {
+        sled_get(&self.db_tree, "revision").map(bytes_to_u32)
```

Comment:
nit: I guess these DB keys should probably be `const` to avoid typos in set vs get. Not going to block this PR on it though.

Reply:
Great first issue! https://github.com/bpfman/bpfman/issues/912